pkg/nuctl/test/project_test.go

```diff
@@ -226,7 +239,71 @@ func (suite *projectExportImportTestSuite) TestImportProject() {
-func (suite *projectExportImportTestSuite) TestImportProjects() {
+func (suite *projectExportImportTestSuite) addUniqueSuffixToImportConfig(configPath, uniqueSuffix string,
+	functionNames, functionEventNames []string, apiGatewayNames []string) string {
+	file, err := ioutil.ReadFile(configPath)
+	suite.Require().NoError(err)
+
+	projectConfig := &command.ProjectImportConfig{}
+	err = yaml.Unmarshal(file, projectConfig)
+	suite.Require().NoError(err)
+
+	projectConfig.Project.Meta.Name = projectConfig.Project.Meta.Name + uniqueSuffix
+	projectConfig.Project.Meta.Namespace = suite.namespace
+	functions := map[string]*functionconfig.Config{}
+	for _, functionName := range functionNames {
+		functionUniqueName := functionName + uniqueSuffix
+		functions[functionUniqueName] = projectConfig.Functions[functionName]
+		functions[functionUniqueName].Meta.Name = functionName + uniqueSuffix
+		functions[functionUniqueName].Meta.Namespace = suite.namespace
+		functions[functionUniqueName].Meta.Labels["nuclio.io/project-name"] =
+			functions[functionUniqueName].Meta.Labels["nuclio.io/project-name"] + uniqueSuffix
+	}
+	projectConfig.Functions = functions
+
+	functionEvents := map[string]*platform.FunctionEventConfig{}
+	for _, functionEventName := range functionEventNames {
+		functionEventUniqueName := functionEventName + uniqueSuffix
+		functionEvents[functionEventUniqueName] = projectConfig.FunctionEvents[functionEventName]
+		functionEvents[functionEventUniqueName].Spec.DisplayName = functionEventName + uniqueSuffix
+		functionEvents[functionEventUniqueName].Meta.Namespace = suite.namespace
+
+		functionEvents[functionEventUniqueName].Meta.Labels["nuclio.io/function-name"] =
+			functionEvents[functionEventUniqueName].Meta.Labels["nuclio.io/function-name"] + uniqueSuffix
+	}
+	projectConfig.FunctionEvents = functionEvents
+
+	apiGateways := map[string]*platform.APIGatewayConfig{}
+	for index, apiGatwayName := range apiGatewayNames {
+		apiGatewayUniqueName := apiGatwayName + uniqueSuffix
+		apiGateways[apiGatewayUniqueName] = projectConfig.APIGateways[apiGatwayName]
+		apiGateways[apiGatewayUniqueName].Meta.Name = apiGatewayUniqueName
+		apiGateways[apiGatewayUniqueName].Meta.Namespace = suite.namespace
+		apiGateways[apiGatewayUniqueName].Spec.Upstreams = []platform.APIGatewayUpstreamSpec{
+			{
+				Kind: platform.APIGatewayUpstreamKindNuclioFunction,
+				Nucliofunction: &platform.NuclioFunctionAPIGatewaySpec{
+					Name: functionNames[index] + uniqueSuffix,
+				},
+			},
+		}
+	}
+	projectConfig.APIGateways = apiGateways
+
+	projectConfigYaml, err := yaml.Marshal(projectConfig)
+	suite.Require().NoError(err)
+
+	// write exported function config to temp file
+	tempFile, err := ioutil.TempFile("", "project-import.*.json")
+	suite.Require().NoError(err)
+
+	_, err = tempFile.Write(projectConfigYaml)
+	suite.Require().NoError(err)
+
+	return tempFile.Name()
+}
+
+func (suite *projectExportImportTestSuite) testImportProjects(apiGatewaysEnabled bool) {
```

Comment:
instead of `apiGatewaysEnabled` you can just inspect the platform name inside and add them when it's kube.
No reason to not test apigateway intentionally unless you can't imo